src/Service/TransactionService.php

```diff
@@ -0,0 +1,304 @@
+use _PhpScoper5eddef0da618a\Mollie\Api\Types\PaymentMethod;
+use _PhpScoper5eddef0da618a\Mollie\Api\Types\PaymentStatus;
+use _PhpScoper5eddef0da618a\Mollie\Api\Types\RefundStatus;
+use Address;
+use Cart;
+use Configuration;
+use Context;
+use Country;
+use Db;
+use Mollie;
+use Mollie\Repository\PaymentMethodRepository;
+use Mollie\Utility\TransactionUtility;
+use Order;
+use OrderPayment;
+use PrestaShop\PrestaShop\Adapter\CoreException;
+use PrestaShopDatabaseException;
+use PrestaShopLogger;
+use Tools;
+
+class TransactionService
+{
+
+    /**
+     * @var Mollie
+     */
+    private $module;
+    /**
+     * @var PaymentMethodRepository
+     */
+    private $paymentMethodRepository;
+    /**
+     * @var \Mollie\Service\OrderStatusService
+     */
+    private $orderStatusService;
+    /**
+     * @var Country
+     */
+    private $country;
+
+    public function __construct(
+        Mollie $module,
+        PaymentMethodRepository $paymentMethodRepository,
+        OrderStatusService $orderStatusService
+    )
+    {
+        $this->module = $module;
+        $this->paymentMethodRepository = $paymentMethodRepository;
+        $this->orderStatusService = $orderStatusService;
+        $this->country = Context::getContext()->country;
+    }
+
+    /**
+     * @param MolliePaymentAlias|MollieOrderAlias $transaction
+     *
+     * @return string|MolliePaymentAlias Returns a single payment (in case of Orders API it returns the highest prio Payment object) or status string
+     *
+     * @throws PrestaShopDatabaseException
+     * @throws PrestaShopException
+     * @throws CoreException
+     * @throws ApiException
+     * @since 3.3.0
+     * @since 3.3.2 Returns the ApiPayment / ApiOrder instead of OK string, NOT OK/NO ID stays the same
+     * @since 3.3.2 Returns the ApiPayment instead of ApiPayment / ApiOrder
+     */
+    public function processTransaction($transaction)
+    {
+        if (empty($transaction)) {
+            if (Configuration::get(Mollie\Config\Config::MOLLIE_DEBUG_LOG) >= Mollie\Config\Config::DEBUG_LOG_ERRORS) {
+                PrestaShopLogger::addLog(__METHOD__.' said: Received webhook request without proper transaction ID.', Mollie\Config\Config::WARNING);
+            }
+
+            return $this->module->l('Transaction failed', 'webhook');
+        }
+
+        // Ensure that we are dealing with a Payment object, in case of transaction ID or Payment object w/ Order ID, convert
+        if ($transaction instanceof MolliePaymentAlias) {
+            if (!empty($transaction->orderId) && TransactionUtility::isOrderTransaction($transaction->orderId)) {
+                // Part of order
+                $transaction = $this->module->api->orders->get($transaction->orderId, array('embed' => 'payments'));
+            } else {
+                // Single payment
+                $apiPayment = $transaction;
+            }
+        }
+
+        if (!isset($apiPayment)) {
+            $apiPayments = array();
```

Comment:
[] instead of array().
Not sure if its your code or its pasted, seems really old :thinking: